runtime/io.c

```diff
@@ -516,6 +526,7 @@ CAMLprim value caml_ml_out_channels_list (value unit)
       Field (res, 0) = chan;
```

Comment:
Good point. I'll move the allocations outside the loop. It may be easiest to maintain a separate count of the open channels. I'll have a go at this next week.